src/checkpoint/container/vector_serialize.h

```diff
@@ -52,13 +52,23 @@
 
 namespace checkpoint {
 
-template <typename Serializer, typename T, typename VectorAllocator>
-void serializeVectorMeta(Serializer& s, std::vector<T, VectorAllocator>& vec) {
+template <typename SerializerT, typename T, typename VectorAllocator>
+typename std::enable_if_t<
+  not std::is_same<SerializerT, checkpoint::Footprinter>::value,
+  void
+> serializeVectorMeta(SerializerT& s, std::vector<T, VectorAllocator>& vec) {
   SerialSizeType vec_size = vec.size();
   s | vec_size;
   vec.resize(vec_size);
 }
 
+template <typename SerializerT, typename T, typename VectorAllocator>
+typename std::enable_if_t<
+  std::is_same<SerializerT, checkpoint::Footprinter>::value,
+  void
+> serializeVectorMeta(SerializerT& s, std::vector<T, VectorAllocator>& vec) { }
+
+
 template <typename Serializer, typename T, typename VectorAllocator>
 void serialize(Serializer& s, std::vector<T, VectorAllocator>& vec) {
   if (s.isFootprinting()) {
```

Comment:
Since `serializeVectorMeta` is now polymorphic on footprinting or not, we could simplify the logic here a bit, to

```
serializeVectorMeta(s, vec);
size_t num_elements = s.isFootprinting() ? vec.capacity() : vec..size();
dispatch::serializeArray(s, &vec[0], num_elements);
```

And have the `s.countBytes(vec)` in the meta function.